File: packages/backend/src/main/kotlin/quizzem/backend/features/game/service/GameService.kt
package quizzem.backend.features.game.service

import org.springframework.stereotype.Service
import quizzem.backend.features.game.api.dto.CreateGameDto
import quizzem.backend.features.game.model.GameModel
import quizzem.backend.features.game.repository.GameRepository
import java.util.*

@Service
class GameService(
    private val gameRepository: GameRepository,
) {
    fun listGames(): List<GameModel> {
        return gameRepository.findAll()
    }

    fun createGame(dto: CreateGameDto): UUID {
        val game = GameModel(title = dto.title, description = dto.description)
        val saved = gameRepository.save(game)
        return requireNotNull(saved.id) { "Game ID was null after save" }
    }
}

File: packages/backend2/src/features/game/game.service.ts
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { CreateGameDto } from '~/src/features/game/dto/create-game.dto';
import { GameModel } from '~/src/features/game/game.model';

@Injectable()
export class GameService {
  private readonly gameRepository: Repository<GameModel>;

  constructor(
    @InjectRepository(GameModel)
    gameRepository: Repository<GameModel>,
  ) {
    this.gameRepository = gameRepository;
  }

  async getGames(): Promise<GameModel[]> {
    return this.gameRepository.find();
  }

  async createGame(createGameDto: CreateGameDto): Promise<GameModel> {
    const newGame = this.gameRepository.create(createGameDto);
    await this.gameRepository.save(newGame);

    return newGame;
  }
}
